builtin/logical/transit/path_decrypt.go

```diff
@@ -44,36 +71,74 @@ func (b *backend) pathDecrypt() *framework.Path {
 
 func (b *backend) pathDecryptWrite(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	name := d.Get("name").(string)
-	ciphertext := d.Get("ciphertext").(string)
-	if len(ciphertext) == 0 {
-		return logical.ErrorResponse("missing ciphertext to decrypt"), logical.ErrInvalidRequest
-	}
-
+	batchInputRaw := d.Get("batch").(string)
+	var batchInputItems []BatchRequestItem
+	var batchInput []byte
 	var err error
-
-	// Decode the context if any
-	contextRaw := d.Get("context").(string)
-	var context []byte
-	if len(contextRaw) != 0 {
-		context, err = base64.StdEncoding.DecodeString(contextRaw)
+	if len(batchInputRaw) != 0 {
+		batchInput, err = base64.StdEncoding.DecodeString(batchInputRaw)
 		if err != nil {
-			return logical.ErrorResponse("failed to base64-decode context"), logical.ErrInvalidRequest
+			return logical.ErrorResponse("failed to base64-decode batch input"), logical.ErrInvalidRequest
+		}
+
+		if err := jsonutil.DecodeJSON([]byte(batchInput), &batchInputItems); err != nil {
+			return nil, fmt.Errorf("invalid input: %v", err)
+		}
+
+	} else {
+		ciphertext := d.Get("ciphertext").(string)
+		if len(ciphertext) == 0 {
+			return logical.ErrorResponse("missing ciphertext to decrypt"), logical.ErrInvalidRequest
+		}
+
+		batchInputItems = make([]BatchRequestItem, 1)
+		batchInputItems[0] = BatchRequestItem{
+			Ciphertext: ciphertext,
+		}
+
+		// Decode the context
+		contextRaw := d.Get("context").(string)
+		if len(contextRaw) != 0 {
+			batchInputItems[0].Context, err = base64.StdEncoding.DecodeString(contextRaw)
+			if err != nil {
+				return logical.ErrorResponse("failed to base64-decode context"), logical.ErrInvalidRequest
+			}
+		}
+
+		// Decode the nonce
+		nonceRaw := d.Get("nonce").(string)
+		if len(nonceRaw) != 0 {
+			batchInputItems[0].Nonce, err = base64.StdEncoding.DecodeString(nonceRaw)
+			if err != nil {
+				return logical.ErrorResponse("failed to base64-decode nonce"), logical.ErrInvalidRequest
+			}
 		}
 	}
 
-	// Decode the nonce if any
-	nonceRaw := d.Get("nonce").(string)
-	var nonce []byte
-	if len(nonceRaw) != 0 {
-		nonce, err = base64.StdEncoding.DecodeString(nonceRaw)
-		if err != nil {
-			return logical.ErrorResponse("failed to base64-decode nonce"), logical.ErrInvalidRequest
+	if len(batchInputItems) == 0 {
```

Comment:
Given that this check will only ever be true if len(batchInputRaw) != 0 (because in the other path we always have size 1) it can probably be moved to the end of that block above.